sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -207,15 +207,23 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                 } else {
                     accumulator = Sets.intersection(accumulator, childIds);
                 }
-
-                // TODO optimisation is to bail here if accumulator is empty
             }
 
             return accumulator;
         }
         if (node instanceof OrQueryNode) {
```

Comment:
Created an FB issue to revisit this later in both Android and iOS.
https://cloudant.fogbugz.com/f/cases/43772/Cloudant-Query-Revisit-executeQueryTree-and-executeSelectorTree-if-else-structure